adobe-private/react-spectrum-v3
# RSP-1837/RSP-1838 Fix console warnings regarding invalid DOM props

ScrollView passes React prop focusedKey as part of {...otherProps} spread to DOM element. We should use filter out focusedKey when declaring variables so only valid props are added to the rendered DOM element.

Popover and Modal pass React props shouldCloseOnBlur and isDismissable as part of {...otherProps} spread to DOM element. We should use filter out shouldCloseOnBlur and isDismissable when declaring variables, so only valid props are added to the rendered DOM element.


Closes [RSP-1837](https://jira.corp.adobe.com/browse/RSP-1837) and [RSP-1838](https://jira.corp.adobe.com/browse/RSP-1838)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding Issues [RSP-1837](https://jira.corp.adobe.com/browse/RSP-1837) and [RSP-1838](https://jira.corp.adobe.com/browse/RSP-1838).

## 📝 Test Instructions:

1. Open Chrome developer console
2. Open http://localhost:9003/?path=/story/modal--default story
3. Verify that expanding Modal does not display react console warning about `isDismissable` prop: `"Warning: React does not recognize the `isDismissable` prop on a DOM element..."`
4. Open http://localhost:9003/?path=/story/picker--default story
5. Verify that expanding Picker does not display react console warning about focusedKey prop: `"Warning: React does not recognize the `focusedKey` prop on a DOM element..."`
6. Open http://localhost:9003/?path=/story/picker--picker-closes-on-blur story
7. Verify that expanding Picker does not display react console warning about `shouldCloseOnBlur` prop: `"Warning: React does not recognize the `shouldCloseOnBlur` prop on a DOM element..."`
8. With same story, reduce width of browser so that Picker expands as a Modal tray.
9. Verify that expanding Picker as a Modal tray does not display react console warning about `shouldCloseOnBlur` prop: `"Warning: React does not recognize the `shouldCloseOnBlur` prop on a DOM element..."`

## 🧢 Your Project:

Accessibility
